src/stream-analytics/azext_stream_analytics/_params.py

```diff
@@ -0,0 +1,180 @@
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+
+    with self.argument_context('stream-analytics output list') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+
+    with self.argument_context('stream-analytics output test') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the output.')
+
+    with self.argument_context('stream-analytics transformation create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the transformation.')
+        c.argument('streaming_units', help='The number of streaming units that the streaming job uses.')
+        c.argument('transformation_query', help='The query that will be run in the streaming job. You can learn more about the Stream Analytics Query Language (SAQL) here: https://msdn.microsoft.com/library/azure/dn834998 .')
+
+    with self.argument_context('stream-analytics transformation update') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the transformation.')
+        c.argument('streaming_units', help='The number of streaming units that the streaming job uses.')
+        c.argument('transformation_query',
+                   help='The query that will be run in the streaming job. You can learn more about the Stream Analytics Query Language (SAQL) here: https://msdn.microsoft.com/library/azure/dn834998 .')
+
+    with self.argument_context('stream-analytics transformation show') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the transformation.')
+
+    with self.argument_context('stream-analytics function create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('job_name', help='The name of the streaming job.')
+        c.argument('name', options_list=['--name', '-n'], help='The name of the function.')
+        c.argument('function_inputs', type=validate_file_or_dict, help='The inputs of the function.')
+        c.argument('function_output', type=validate_file_or_dict, help='The output of the function.')
+        c.argument('function_binding', type=validate_file_or_dict, help='The binding of the function.')
```

Comment:
remove function_ prefix from above 3 since redundant.

Reply:
same as `query`, `output` will have conflict problem.

Reply:
got it. how about the other 2

Reply:
function_inputs -> inputs, function_binding -> binding.